tests/Unit/ComposerConfigProcessTest.php

```diff
@@ -75,8 +75,112 @@ public function testProcessWithPackagesForCheckAndWithPresenceControlledPackage(
     {
+        $composer = $this->createComposerMock([
+            'config-plugin-options' => [
+                'source-directory' => 'custom-dir',
+            ],
+            'config-plugin' => [
+                'common' => 'subdir/*.php',
+                'not-exists' => '?not-exists/*.php',
+                'params' => [
+                    'params.php',
+                    '?params-local.php',
+                ],
+                'web' => [
+                    '$common',
+                    'web.php',
+                ],
+            ],
+            'config-plugin-environments' => [
+                'alfa' => [
+                    'params' => 'alfa/params.php',
+                    'web' => 'alfa/web.php',
+                    'main' => [
+                        '$web',
+                        'alfa/main.php'
+                    ],
+                ],
+                Options::DEFAULT_ENVIRONMENT => [
+                    'params' => 'alfa/params.php',
+                    'web' => 'alfa/web.php',
+                    'main' => [
+                        '$web',
+                        'alfa/main.php'
+                    ],
+                ],
+            ],
+        ]);
+
+        $process = new ComposerConfigProcess($composer, [], $forceCheck);
+
+        $this->assertProcessData($process, $forceCheck, [
+            Options::DEFAULT_ENVIRONMENT => [
+                'common' => [
+                    '/' => [
+                        'custom-dir/subdir/*.php',
+                    ],
+                    'test/custom-source' => [
+                        'subdir/*.php',
+                    ],
+                ],
+                'not-exists' => [
+                    '/' => [
+                        '?custom-dir/not-exists/*.php',
+                    ],
+                ],
+                'params' => [
+                    '/' => [
+                        'custom-dir/params.php',
+                        '?custom-dir/params-local.php',
+                    ],
+                    'test/custom-source' => [
+                        'params.php',
+                    ],
+                ],
+                'web' => [
+                    '/' => [
+                        '$common',
+                        'custom-dir/web.php',
+                    ],
+                    'test/custom-source' => [
+                        '$common',
+                        'web.php',
+                    ],
+                ],
+            ],
+            'alfa' => [
+                'main' => [
+                    Options::DEFAULT_ENVIRONMENT => [
+                        '$web',
+                        'alfa/main.php',
+                    ],
+                ],
+                'params' => [
+                    Options::DEFAULT_ENVIRONMENT => [
+                        'alfa/params.php',
+                    ],
+                ],
+                'web' => [
+                    Options::DEFAULT_ENVIRONMENT => [
```

Comment:
Seems here `Options::DEFAULT_ENVIRONMENT` not environment. This is application files for config. May be create new constant for this?